core/src/test/scala/org/apache/spark/FutureActionSuite.scala

```diff
@@ -49,4 +50,20 @@ class FutureActionSuite
     job.jobIds.size should be (2)
   }
 
+  test("simple async action callbacks should not tie up execution context threads (SPARK-9026)") {
+    val rdd = sc.parallelize(1 to 10, 2).map(_ => Thread.sleep(1000 * 1000))
+    val pool = ThreadUtils.newDaemonCachedThreadPool("SimpleFutureActionTest")
+    val executionContext = ExecutionContext.fromExecutorService(pool)
+    val job = rdd.countAsync()
+    try {
+      for (_ <- 1 to 10) {
+        job.onComplete(_ => ())(executionContext)
+        assert(pool.getLargestPoolSize < 10)
```

Comment:
I think that my intention when writing this test was to have a test that demonstrated the eagerly-create-a-thread-per-callback problem with the old implementation of SimpleFutureAction.

I don't think that this is flaky but I also don't think that this tests adds much value since we're unlikely to ever switch back to the old inefficient implementation.  I'll just drop this test, since I don't think it's adding any real value right now.